feat(core): update ban on Vo

bodnara opened the pull request:
* updateBan for VoManager was added to missing layers
* created test for updateBan on Vo
* setBan no longer updates existing ban but throws exception to stay consistent with variants of this method for other objects

xflord commented on perun-core/src/main/java/cz/metacentrum/perun/core/entry/VosManagerEntry.java:
```diff
@@ -694,6 +696,23 @@ public BanOnVo setBan(PerunSession sess, BanOnVo ban) throws PrivilegeException,
+		Vo vo = getPerunBl().getVosManagerBl().getVoById(sess, banOnVo.getVoId());
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "updateBan_BanOnVo_policy", Arrays.asList(banOnVo, member, vo))) {
```
```suggestion
		if (!AuthzResolver.authorizedInternal(sess, "updateBan_BanOnVo_policy", Arrays.asList(banOnVo))) {
```
Vo and Member get fetched from BanOnVo, no need to include them here

bodnara commented on perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/VosManagerBlImpl.java:
```diff
@@ -872,15 +873,17 @@ public List<BanOnVo> getBansForVo(PerunSession sess, int voId) {
-		return vosManagerImpl.updateBan(sess, banOnVo);
+		banOnVo = getVosManagerImpl().updateBan(sess, banOnVo);
+		Member member = getPerunBl().getMembersManagerBl().getMemberById(sess, banOnVo.getMemberId());
+		getPerunBl().getAuditer().log(sess, new MemberSuspended(member));
```
I am not sure either, I assumed it just generates log messages.

bodnara replied:
I created new event for the ban update after all.

Kuliak commented on perun-core/src/test/java/cz/metacentrum/perun/core/entry/VosManagerEntryIntegrationTest.java:
```diff
@@ -558,6 +559,26 @@ public void getBansForVo() throws Exception {
+
+		BanOnVo banOnVo = new BanOnVo();
+		banOnVo.setMemberId(member.getId());
+		banOnVo.setVoId(vo.getId());
```
Technically, you don't have to set VO's ID here because it is set later in the setBan method using member's voId value. However, it's just a tiny detail.

Kuliak reviewed (commented):
I like the solution, good job.
There are some unused imports (as a leftover from some other commits as well) in files that might have been removed as a cleanup. You can have it done automatically as well, so you don't have to solve this problem in the future manually.

Maybe just a small nitpick: Since we use BanAlreadyExists exception for VO, we might want to update its javadoc that says it is used for facility/resource only.

bodnara commented:
> I like the solution, good job. There are some unused imports (as a leftover from some other commits as well) in files that might have been removed as a cleanup. You can have it done automatically as well, so you don't have to solve this problem in the future manually.

Removed unused imports from affeted files.

> Maybe just a small nitpick: Since we use BanAlreadyExists exception for VO, we might want to update its javadoc that says it is used for facility/resource only.

Updated javadoc.

xflord commented on perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/VosManagerBlImpl.java:
```diff
@@ -65,17 +64,7 @@
-import java.util.Map;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
```
Please no wildcard imports

Kuliak commented on perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/VosManagerBlImpl.java:
```diff
@@ -65,17 +64,7 @@
-import java.util.Map;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
```
we don't want to use wildcard imports. Please, use explicit imports without asterisk
(can be set in: **_Settings - Editor - Code Style - Java - Imports - Class count to use import with '*': 500_**)

bodnara replied:
Thanks for the settings tip.
